### ISSUE TITLE
instantill response from subscriptions flaky

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1766]
I think this happened since we pushed the crossref updates.

You can see this happening in https://app.ghostinspector.com/tests/5f4e8ba113246d7fa0af59b9.

When I go to the page I can get it to succeed after a hard refresh, but not always.

**Comment by markmacgillivray** [comment]
I've confirmed I can occasionally make it fail as well. However checking a successful and an unsuccessful one, it's not us. We are scraping and sending identical data to an identical serialssolutions URL and getting back a different result in some cases. I don't know why, and I don't think there is anything we can do about it.

I can't think of anything else to check, and trying it manually about 50 times I only got the wrong result back once. Is it worth doing more exploration of their system? Or shall we close this one now?

**Comment by JosephMcArthur** [comment]
Can you send me the seriaksolutions URL / query we're making? I will likely
ask them if this is a known issue.

**Comment by markmacgillivray** [comment]
Sent

**Comment by JosephMcArthur** [comment]
When I hit the SS url you sent directly, I've failed to see it loss `<ssopenurl:url type="article">` (in around 30 attempts), but I see our response lack the expected full text around 50% of the time. The difference there still makes me suspect this is on our end, as does the timing of this starting right around when we deployed other changes.

Does our system wait for SS to return before showing a response?

**Comment by markmacgillivray** [comment]
Yes I also did it myself many times from the browser and it always returned correctly. Like 100 times while doing other work. But when I do it from our server directly, not via the app itself, it fails about half the time. So it's not to do with changes to our code but could be to do with where we send from. I could route it through puppeteer maybe, although it is pretty weird for that to matter in an XML response context. Perhaps they have some sort of restriction going on based on programmatic access though. It is easy to change it to use puppeteer, but it will make it slower.

**Comment by JosephMcArthur** [comment]
Okay, this is starting to make more sense. So, their end is flakey when called via a server.

Couple of Q's. How hard is it for us to try some queries via puppeteer to see if that resolves the issue? How much slower will it make it?

Joseph McArthur
Assistant Director: Right to Research Coalition ( http://righttoresearch.org )
Director / Founder: Open Access Button ( http://openaccessbutton.org )
Twitter: @Mcarthur_Joe ( https://twitter.com/Mcarthur_Joe )
Skype: joseph_mcarthur

**Comment by markmacgillivray** [comment]
unfortunately I tried putting it through puppeteer and still got back the page missing the necessary element. It's pretty weird - if it was some sort of timed delay or outright block on our server IP, that would make sense. I don't know what they would be doing that would cause us to sometimes get part of a result page and other times not get that part. There's no reason why it should even matter.

The only other thing we could try is to start faking request headers or routing through multiple IPs. However since it doesn't really appear to be an attempt to block or limit us, that may not help. Anyway that's usually the last resort, would be nicer if we can find out from them why this would be happening.

**Comment by JosephMcArthur** [comment]
Okay, I am going to dive into the process of trying to resolve with them.

**Comment by markmacgillivray** [comment]
OK

**Comment by markmacgillivray** [comment]
Is there anything else I can do with this one? Looks like it's with @JosephMcArthur for follow-up with customer so I will change the assignment for now.
